### PR TITLE
feat(kit): `InputYear` add new component

### LINKED ISSUE #2745: 🚀 - InputYear
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

_No response_

### Description

Add please component `InputYear` with support in the version 2.x

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2745

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/app/pages.ts [hunk 1]
```diff
@@ -570,6 +570,12 @@ export const pages: TuiDocPages = [
                 keywords: `поле, инпут, форма, ввод, input, time, hour, minute, время, час, минута`,
                 route: `/components/input-time`,
             },
+            {
+                section: $localize`Components`,
+                title: `InputYear`,
+                keywords: `поле, инпут, форма, ввод, input, month, месяц, год, дата`,
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/app/pages.ts:576]: nit: "month месяц" is redundant?

File: projects/demo/src/modules/components/input-year/examples/2/index.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        class="tui-space_bottom-2"
+    >
+        Choose a month
+    </tui-input-year>
+    <tui-input-year
+        formControlName="testValue"
+        tuiTextfieldSize="m"
+        class="tui-space_bottom-2"
+    >
+        Choose a month
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/input-year/examples/2/index.html:17]: choose a year

File: projects/demo/src/modules/components/input-year/examples/import/insert-template.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+```html
+<form [formGroup]="testForm">
+  <tui-input-year formControlName="testValue">Choose a month</tui-input-year>
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/input-year/examples/import/insert-template.md:3]: choose a year

File: projects/demo/src/modules/components/input-year/input-year.template.html [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                    [pseudoFocus]="pseudoFocused"
+                    [pseudoHover]="pseudoHovered"
+                    [pseudoActive]="pseudoPressed"
+                    [disabledItemHandler]="disabledItemHandler"
+                    [tuiTextfieldIconLeft]="iconLeft"
+                    [tuiTextfieldCleaner]="cleaner"
+                    [tuiTextfieldLabelOutside]="labelOutside"
+                    [tuiTextfieldSize]="size"
+                >
+                    Month
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/input-year/input-year.template.html:54]: year?

File: projects/kit/components/input-year/input-year.component.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Input()
+    @tuiDefaultProp()
+    disabledItemHandler: TuiBooleanHandler<number> = ALWAYS_FALSE_HANDLER;
+
+    open = false;
+
+    readonly initialItem = new Date().getFullYear();
+
+    readonly textMaskOptions = {
```
Comment (vladimirpotekhin) [inline projects/kit/components/input-year/input-year.component.ts:62]: do we really need a mask? how about just force textfield length to 4?

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**